firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -119,11 +120,14 @@ public Thread newThread(Runnable r) {
       RandomFidGenerator fidGenerator) {
     this.firebaseApp = firebaseApp;
     this.serviceClient = serviceClient;
-    this.executor = executor;
     this.persistedInstallation = persistedInstallation;
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    this.backgroundExecutor = backgroundExecutor;
+    this.networkExecutor =
+        new ThreadPoolExecutor(
+            1, 1, 0L, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(), threadFactory);
```

Comment:
Executors.newCachedThreadPool() would be good here.

 If you choose to explicitly create it, please document the constants.

Reply:
As mentioned earlier, reason for explicitly creating is some 1p apps restrict creating unnamed thread pools. We got the ThreadFactory created in FIS whitelisted.

Refer to b/148952570

I documented the constants.